### PR TITLE
[Storage] Support HNS soft delete

### PR DESCRIPTION
**Pull request by yifanz0**
## Description

Design passed review in https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/813

Add support for HNS soft delete. 

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+
+   FileSystem Name: filesystem1
+
+Path                 DeletionId           DeletedOn            RemainingRetentionDays
+----                 ----------           ---------            ----------------------
+dir0/dir1/file1      132658816156507617   2021-05-19 07:06:55Z 3                     
+dir0/dir2            132658834541610122   2021-05-19 07:37:34Z 3                    
+dir0/dir2/file3      132658834534174806   2021-05-19 07:37:33Z 3
+```
```
Comment (blueww) [inline src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md:39]: Please change the same to following format, or the Static analysis might fail:

````markdown
```powershell
$localUser = Get-AzStorageLocalUser ..... (note: "PS C:|>" needs to be deleted)
```

```output
......
```
````

Comment (blueww) [reply src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md:39]: same as below examples

File: src/Storage/Storage.Management/help/Restore-AzDataLakeGen2DeletedItem.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+PS C:\> $items | Restore-AzDataLakeGen2DeletedItem 
+
+   FileSystem Name: filesystem1
+
+Path                 IsDirectory  Length          LastModified         Permissions  Owner                Group               
+----                 -----------  ------          ------------         -----------  -----                -----               
+dir0/dir1/file1      False        1024            2021-05-19 07:06:39Z rw-r-----    $superuser           $superuser          
+dir0/dir2            True                         2021-05-19 07:06:37Z rwxr-x---    $superuser           $superuser          
+dir0/dir2/file3      False        1024            2021-05-19 07:06:42Z rw-r-----    $superuser           $superuser
+```
```
Comment (blueww) [inline src/Storage/Storage.Management/help/Restore-AzDataLakeGen2DeletedItem.md:58]: This example has multiple output, might need add 
`<!-- Skip: Output cannot be splitted from code -->`

File: src/Storage/Storage/DatalakeGen2/Cmdlet/RestoreAzDataLakeGen2DeletedItem.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        [Parameter(Mandatory = true, HelpMessage = "Azure Datalake Gen2 Deleted Item Object to restore.",
+            ValueFromPipeline = true, ParameterSetName = ItemParameterSet)]
+        [ValidateNotNull]
+        public AzureDataLakeGen2DeletedItem InputObject { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        //[Parameter(Mandatory = false, HelpMessage = "Return the restored item.")]
+        //public SwitchParameter PassThru { get; set; }
```
Comment (blueww) [inline src/Storage/Storage/DatalakeGen2/Cmdlet/RestoreAzDataLakeGen2DeletedItem.cs:65]: Please delete the 2 commented lines.

Comment (yifanz0) [reply src/Storage/Storage/DatalakeGen2/Cmdlet/RestoreAzDataLakeGen2DeletedItem.cs]: Done

File: src/Storage/Storage.Management/help/Restore-AzDataLakeGen2DeletedItem.md (hunk 2 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/help/Restore-AzDataLakeGen2DeletedItem.md:58]: Done

File: src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md (hunk 1 above)
Comment (yifanz0) [reply src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md:40]: Done

File: src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md [hunk 4]
```diff
@@ -25,11 +25,12 @@ This cmdlet only works if Hierarchical Namespace is enabled for the Storage acco
 ## EXAMPLES
 
 ### Example 1: List all deleted files or directories from a Filesystem
-```
+```powershell
 PS C:\> Get-AzDataLakeGen2DeletedItem -FileSystem "filesystem1"
```
Comment (blueww) [inline src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md:29]: remove "PS C:\> "

Comment (yifanz0) [reply src/Storage/Storage.Management/help/Get-AzDataLakeGen2DeletedItem.md]: Done

**Review by vidai-msft — APPROVED** [review]
Looks good

**Pull request merged**